airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/http_requester.py

```diff
@@ -258,7 +258,13 @@ def _request_headers(
         )
         if isinstance(headers, str):
             raise ValueError("Request headers cannot be a string")
-        return {str(k): str(v) for k, v in headers.items()}
+        formatted_headers = {}
```

Comment:
I'm almost wondering if we should have this assumption for the user. I mean, if the user wants to comma-separate its value, that's fine. It feels different than the params as params will induce a duplication of the keys and therefore, we need to know if the value is an array. Here, the fact that it's a list seems more like an API decision between the client and the server as to how they interpret the string than something that impact how we create HTTP requests. In other words, it feels safer to do a little formatting/changes as possible to this value as it end up just being a string in terms of outgoing HTTP request. I fear that manipulation might create a case where the user has typed something but we change it because of the formatting we're trying to do. Does that make sense?

Reply:
@maxi297 just to be clear, are you suggesting that we don't turn the value into a list if it looks like one?

If so, then how would the user achieve the `a=1&a=2&a=3` behavior? Maybe we should allow them to set the same query parameter key multiple times? (currently the last occurrence of a key overrides any other occurrences of that same key)

Reply:
I think this code is only for headers. For param, `requests` should handle this under the hood

Reply:
Ah, I didn't realize that when I wrote the above comment that this was for headers only. I think for headers I'm on the same page as you, let's do as little "magic" as possible until we get user requests to add more

Reply:
Makes sense to me. Removed.